worker-document/src/main/java/com/hpe/caf/worker/document/scripting/specs/UrlScriptSpec.java

```diff
@@ -43,10 +44,11 @@ public UrlScriptSpec(final URL url) throws URISyntaxException
     @Override
     public CompiledScript compile(final Compilable compiler) throws ScriptException
     {
-        // This method has been overridden in order to avoid calling close() on the URLReader after the compile() call.
-        // Doing so actually causes it to read from the URL a further time!
-        final Reader reader = new URLReader(url);
-        return compiler.compile(reader);
+        try (final Reader reader = new BufferedReader(new InputStreamReader(url.openStream()))) {
+            return compiler.compile(reader);
+        } catch (final IOException e) {
+            throw new ScriptException(e);
+        }
```

Comment:
With these changes this method is now no different to the method it is overriding.  As such it should simply be removed.  Overriding it no longer makes sense.